README.md

```diff
@@ -18,103 +18,67 @@
-* `PORT` -- default `8080`
-* `LISTEN_FD` -- if supplied on unix-like systems, if this is set to an open file descriptor number, the server will listen to that fd
-* `OTEL_EXPORTER_PROMETHEUS_HOST` -- default `"localhost"`
-* `OTEL_EXPORTER_PROMETHEUS_PORT` -- default `9464`
-* `ASSET_DIR` -- set this to skip building the react app and include react app assets from a different directory
-
-## Initial setup
-
-### Migrating the database
-
-First, create the database referred to by `DATABASE_URL` in your environment. This may an invocation of [`createdb`](https://www.postgresql.org/docs/current/app-createdb.html) if running postgresql locally, or postgres inside of a docker container.
-
-`cargo run -p migration -- up` will bring the application up to the current schema.
-For more options, execute `cargo run -p migration -- --help`.
+1. Clone the repository and navigate to its root.
+1. Execute `docker-compose watch`.
+1. Navigate in your browser to `http://app.divviup.local:8080/`.
 
-### Installing npm dependencies
+`docker-compose` will automatically reload containers when you make changes. Data is persisted
+until you `docker-compose rm --volumes`.
 
-```bash
-$ cd app && npm ci && cd -
-```
+Two Janus aggregators will be created for you, but are not automatically paired to divviup-api.
+Their information is:
+1. Address: `http://janus_1_aggregator:8080/aggregator-api`, Token: `0000`
+1. Address: `http://janus_2_aggregator:8080/aggregator-api`, Token: `0000`
 
-## Running the server
+If you need to talk to these aggregators from outside docker-compose's network namespace, e.g.
+with a testing client, they are mapped to `localhost:9001` and `localhost:9002`, respectively.
 
-This service has dependencies on several external services. In order to support development and testing, there is an `api-mocks` cargo feature to stub out all external services including aggregator apis, auth0, and postmark.
+PostgreSQL is exposed on port 5432 with username and password `postgres`.
 
-As such, to run a standalone development server,
+If you need to iterate on database migrations, you may wish to disable the `divviup_api_migrate`
+service by commenting it out in `compose.yaml`.
 
-```bash
-$ cargo run --features api-mocks
-```
+### Faster frontend iteration
 
-### Embedded React App
+For iterating on the frontend, docker-compose can be slow since the divviup-api backend needs to
+be recompiled for every frontend change. Instead, run the frontend separately from the backend to
+take advantage of the speed of [Vite](https://vitejs.dev/).
 
-If an environment variable `ASSET_DIR` is available, all files in that directory will be served as a virtual host on `APP_URL`.
+1. Create the `api_url` file so the app knows where to hit the API.
+    ```bash
+    echo "http://api.divviup.local:8080/" >app/public/api_url
+    ```
+2. Run the backend with the `no-app` override:
+    ```bash
+    docker-compose -f compose.yaml -f compose.no-app.yaml watch
```

Comment:
As a future enhancement, it might be nice to instead provide an override that builds a container that just has NPM dependencies installed, and runs the vite dev server on a bind-mounted source directory.

Reply:
I like this. We can use this as the main way of development, since it allows for some simplifications (don't need /etc/hosts hacks for vhost routing), and I think we'd always prefer front-end iteration to be fast.

Reply:
Ah, regrettably to get hot reload to work on MacOS with Colima, it's a bit involved. This should be a net positive for everyone else on Linux though, so I'll go forward with this.